spec-dtslint/operators/concatMap-spec.ts

```diff
@@ -33,6 +33,10 @@ it('should support union-type projections', () => {
   const o = of(Math.random()).pipe(concatMap(n => n > 0.5 ? of('life') : of(42))); // $ExpectType Observable<string | number>
 });
 
+it('should support union-type projections with empty streams', () => {
+  const o: Observable<number> = of(1, 2, 3).pipe(concatMap(n => Math.random() < 0.5 ? of(123) : of()));
```

Comment:
This needs to be written as:

```ts
const o = of(1, 2, 3).pipe(concatMap(n => Math.random() < 0.5 ? of(123) : of())); // $ExpectType Observable<number>
```

dtslint doesn't care about or report compilation errors. An expectation is required, otherwise dtslint would pass this even if the inferred return type wasn't assignable to `Observable<number>`.